pandas/_libs/index.pyx

```diff
@@ -72,9 +72,10 @@ cdef class IndexEngine:
         self.over_size_threshold = n >= _SIZE_CUTOFF
         self.clear_mapping()
 
-    def __contains__(self, object val):
+    def __contains__(self, val: object) -> bool:
+        # We assume before we get here:
```

Comment:
can val be typed as Hashable, (comment then not needed?)

Reply:
im planning to add more assumptions to the comment in future PRs

Reply:
you *could* add an assert